tests/python_api/python_function_example.py

```diff
@@ -0,0 +1,125 @@
+    
+    def __call__(self, batch):
+        # Normalize using user passed mean and std
+        return ((batch - self.mean) / self.std).astype(np.float32)  # Casting is needed since it will return fp64 outputs otherwise
+
+def draw_patches(image, idx, layout="nchw", dtype="fp32", device="cpu"):
+    # image is expected as a numpy array
+    if layout == "nchw":
+        image = image.transpose([1, 2, 0])
+    if dtype in ["fp16", "fp32"]:
+        image = image.astype("uint8")
+    image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+    cv2.imwrite(f"output_folder/python_function/{idx}_train.png", image*255)
+
+
+def main():
+    if len(sys.argv) < 3:
+        print("Please pass image_folder batch_size")
+        exit(0)
+    try:
+        path = "output_folder/python_function/"
+        isExist = os.path.exists(path)
+        if not isExist:
+            os.makedirs(path)
+    except OSError as error:
+        print(error)
+    data_path = sys.argv[1]
+    rocal_cpu = True  # Only supported for Host backend
+    batch_size = int(sys.argv[2])
+    random_seed = random.SystemRandom().randint(0, 2**32 - 1)
+
+    local_rank = 0
+    world_size = 1
+    normalizer = NormalizeWithStats(mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+                                        std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    crop_image_fn = partial(crop_fn, crop_size=(224, 224))
+    # Pipeline example with random brightness + crop + hflip + normalize
+    pipe = Pipeline(batch_size=batch_size, num_threads=8, device_id=local_rank,
+                                                   seed=random_seed, rocal_cpu=rocal_cpu, tensor_layout=types.NHWC, tensor_dtype=types.FLOAT16)
+    with pipe:
+        jpegs, _ = fn.readers.file(file_root=data_path)
+        decode = fn.decoders.image(jpegs, file_root=data_path, output_type=types.RGB, shard_id=local_rank, num_shards=world_size, random_shuffle=False)
+        rand_brightness_output = fn.python_function(decode, function = brightness_fn, dtype=types.UINT8, layout=types.NHWC)
+        cropped_output = fn.python_function(rand_brightness_output, function = crop_image_fn, output_dims=(224, 224, 3), dtype=types.UINT8, layout=types.NHWC)
+        flipped_output = fn.python_function(cropped_output, function = flip_fn, dtype=types.UINT8, layout=types.NHWC)
+        normalized_output = fn.python_function(flipped_output, function = normalizer, dtype=types.FLOAT, layout=types.NHWC)
+        pipe.set_outputs(normalized_output)
+    pipe.build()
+    
+    # Dataloader
+    data_loader = ROCALClassificationIterator(pipe, device="cpu", device_id=local_rank)
+    cnt = 0
+
+    # Enumerate over the Dataloader
+    for epoch in range(3):
+        print(
+            "+++++++++++++++++++++++++++++EPOCH+++++++++++++++++++++++++++++++++++++", epoch)
+        for i, it in enumerate(data_loader):
+            print(
+                "************************************** i *************************************", i)
```

Comment:
what is the use of printing this line?

Reply:
I think this is unnecessary to print for every iteration. We need to make another PR for cleaning this up for all test cases